src/python/pants/util/objects.py

```diff
@@ -164,6 +159,11 @@ def __str__(self):
         class_name=type(self).__name__,
         typed_tagged_elements=', '.join(elements_formatted))
 
+    def copy(self, **kwargs):
```

Comment:
The `_replace` method above is a non-typesafe api-copy of copy from namedtuple. It has no production callers - can it be killed? If not, can it just be an alias for `_replace` or vice versa?

Reply:
I addressed this in 5c13575! Is it considered to be breaking the API if we now raise a `TypeError` on a field name that doesn't exist, instead of the `ValueError` of the previous `_replace()` implementation? I would think not, but I'm not sure.

Reply:
It would be a debatable break since that part of the API was not documented. The `datatype` function is not marked `:API: public` though (see: https://www.pantsbuild.org/deprecation_policy.html), so this is technically moot.